### PR TITLE
Tabs for schedule management dialog

### PR DESCRIPTION
**Pull request by oleg-morozov**
JIRA: TNT-517

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7445/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9297/ : FAILURE in 8m 14s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7447/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9299/ : SUCCESS in 13m 50s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7452/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9304/ : SUCCESS in 13m 45s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7453/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9305/ : SUCCESS in 16m 00s

File: libs/sdk-ui-dashboard/src/presentation/scheduledEmail/DefaultScheduledEmailManagementDialog/ScheduledEmailManagementDialog.tsx [hunk 1]
```diff
@@ -2,13 +2,16 @@
...
 import { IScheduledEmailManagementDialogProps } from "../types";
 import { ScheduledEmails } from "./ScheduledEmails";
-import { useScheduledEmailManagement } from "./useScheduledEmailManagement";
+import { useScheduledEmailManagement, IScheduledManagementResult } from "./useScheduledEmailManagement";
 import { DeleteScheduleConfirmDialog } from "./DeleteScheduleConfirmDialog";
-import { selectCurrentUser, useDashboardSelector } from "../../../model";
+import { selectCurrentUser, useDashboardSelector, selectCanManageScheduledMail } from "../../../model";
+
+const BY_USER_TAB_ID = "gs.visualizationsList.tabs.my";
+const ALL_TAB_ID = "gs.visualizationsList.tabs.all";
```
Comment (scavnickyj) [inline libs/sdk-ui-dashboard/src/presentation/scheduledEmail/DefaultScheduledEmailManagementDialog/ScheduledEmailManagementDialog.tsx:14]: Why are you using `gs.visualizationsList.tabs`? Don't we have full control over these ids? Something like `scheduledEmailsList.tabs` would be more suitable..

Comment (scavnickyj) [reply libs/sdk-ui-dashboard/src/presentation/scheduledEmail/DefaultScheduledEmailManagementDialog/ScheduledEmailManagementDialog.tsx:14]: Now I see that these are translation ids... nevermind.

File: libs/sdk-ui-dashboard/src/presentation/scheduledEmail/DefaultScheduledEmailManagementDialog/ScheduledEmailManagementDialog.tsx [hunk 2]
```diff
@@ -36,30 +50,49 @@ export const ScheduledEmailManagementDialog: React.FC<IScheduledEmailManagementD
...
+    }, []);
+
     useScheduledEmailManagement({
         loadScheduledMails: isLoading,
         onError: onLoadError,
         onSuccess: onLoadSuccess,
     });
 
+    const scheduledEmails = selectedTabId === ALL_TAB_ID ? scheduledEmailsAll : scheduledEmailsByUser;
+    const messageForEmptyId =
```
Comment (scavnickyj) [inline libs/sdk-ui-dashboard/src/presentation/scheduledEmail/DefaultScheduledEmailManagementDialog/ScheduledEmailManagementDialog.tsx:64]: I would rename to `noSchedulesMessageId`

File: libs/sdk-ui-dashboard/src/presentation/scheduledEmail/DefaultScheduledEmailManagementDialog/ScheduledEmailManagementDialog.tsx [hunk 3]
```diff
@@ -36,30 +50,49 @@ export const ScheduledEmailManagementDialog: React.FC<IScheduledEmailManagementD
...
+                        onTabSelect={handleTabChange}
+                    />
+                )}
                 <div className="gd-scheduled-emails-content">
                     <ScheduledEmails
                         onDelete={handleScheduleDelete}
                         isLoading={isLoading}
                         scheduledEmails={scheduledEmails}
                         currentUserEmail={currentUser?.email}
+                        messageForEmpty={<FormattedMessage id={messageForEmptyId} values={{ br: <br /> }} />}
```
Comment (scavnickyj) [inline libs/sdk-ui-dashboard/src/presentation/scheduledEmail/DefaultScheduledEmailManagementDialog/ScheduledEmailManagementDialog.tsx:95]: Can we just pass the id and handle the translation in the child component? I would also name the prop accordingly like `noSchedulesMessageId`

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7462/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9314/ : SUCCESS in 13m 45s

**Comment by scavnickyj** [comment]
extended test

**Review by scavnickyj — APPROVED** [review]

**Comment by scavnickyj** [comment]
LGTM

**Comment by oleg-morozov** [comment]
extended check cypress

**Comment by oleg-morozov** [comment]
extended check sonar

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7464/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/192/ : SUCCESS in 12m 11s

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-check-extended-sonar-PR2400

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7465/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1266/ : SUCCESS in 14m 44s

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1441/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7463/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1441/ : UNSTABLE in 22m 20s

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2400

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2158/ : SUCCESS in 1m 06s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2444/ : SUCCESS in 22s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1267/ : SUCCESS in 17m 27s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9318/ : SUCCESS in 14m 06s

**Pull request merged**